yarn-project/simulator/src/avm/opcodes/hashing.ts

```diff
@@ -0,0 +1,138 @@
+import { toBigIntBE } from '@aztec/foundation/bigint-buffer';
+import { keccak, pedersenHash, poseidonHash, sha256 } from '@aztec/foundation/crypto';
+
+import { AvmContext } from '../avm_context.js';
+import { Field } from '../avm_memory_types.js';
+import { Opcode, OperandType } from '../serialization/instruction_serialization.js';
+import { Instruction } from './instruction.js';
+
+export class Poseidon2 extends Instruction {
+  static type: string = 'POSEIDON2';
+  static readonly opcode: Opcode = Opcode.POSEIDON;
+
+  // Informs (de)serialization. See Instruction.deserialize.
+  static readonly wireFormat: OperandType[] = [
+    OperandType.UINT8,
+    OperandType.UINT32,
+    OperandType.UINT32,
+    OperandType.UINT32,
+  ];
+
+  constructor(private destOffset: number, private hashOffset: number, private hashSize: number) {
+    super();
+  }
+
+  async execute(context: AvmContext): Promise<void> {
+    // We hash a set of field elements
+    const hashData = context.machineState.memory
+      .getSlice(this.hashOffset, this.hashOffset + this.hashSize)
+      .map(word => word.toBuffer());
+
+    const hash = poseidonHash(hashData);
+    context.machineState.memory.set(this.destOffset, new Field(hash));
+
+    context.machineState.incrementPc();
+  }
+}
+
+export class Keccak extends Instruction {
+  static type: string = 'KECCAK';
+  static readonly opcode: Opcode = Opcode.KECCAK;
+
+  // Informs (de)serialization. See Instruction.deserialize.
+  static readonly wireFormat: OperandType[] = [
+    OperandType.UINT8,
+    OperandType.UINT32,
+    OperandType.UINT32,
+    OperandType.UINT32,
+  ];
+
+  constructor(private destOffset: number, private hashOffset: number, private hashSize: number) {
+    super();
+  }
```

Comment:
`destOffset` -> `dstOffset`, and make it the last operand (to match other instructions)